### ISSUE TITLE
Async vault usage option. 

### ISSUE THREAD

**Issue by roboalex2** [PikaMug/Quests#1349]
Please add async vault support if you have time. It would be nice to switch between sync and async vault in the config. 

Many money plugins perform SQL queries upon money change. Most of them support async usage. 

The option to use the vault api async could prevent short lag spikes resulting from SQL queries or similar
slow operation performed by the money plugin on the main thread.

**Comment by PikaMug** [comment]
Thanks for your request. As far as I can tell, Vault doesn't offer any kind of asynchronous API. Whether this is because the developers do not care to, or because not all of its operations are thread safe, I cannot say. However, as Quests does not check balances on a regular basis (mainly only to issue rewards and check requirements), the benefits would be minimal. Therefore, I am not interested in looking into this further unless you have an error report or evidence of _significant_ time savings.

**Comment by roboalex2** [comment]
Vault just passes the call to the underlaying money plugin. Some of them support async call's some don't. Depending on the money plugin used it can take less than a millisecond or up to half a second. 

I wrote this because I noticed every time a player gained money by fulfilling a quest a short lagspike (150ms) was visible on our tps monitoring tool. On its own that's not much, but players could exploit this to drop the server performance.
